railties/lib/rails/application/default_middleware_stack.rb

```diff
@@ -34,7 +34,7 @@ def build_stack
           # Must come after Rack::MethodOverride to properly log overridden methods
           middleware.use ::Rails::Rack::Logger, config.log_tags
           middleware.use ::ActionDispatch::ShowExceptions, show_exceptions_app
-          middleware.use ::ActionDispatch::DebugExceptions, app
+          middleware.use ::ActionDispatch::DebugExceptions, app, config.api_only
```

Comment:
:-1: on using `api_only` here.

I'm in two minds as to whether we should just always give a JSON error for a request that wants JSON... but if we want to keep the current HTML-for-JSON behaviour for "regular" applications, this should be based on a new config setting (which would get set to true by the API generator).

If "make Rails better for APIs" means "introduce some new configurable features, and some new defaults", we'll all win. But if it means "introduce an expanding number of places that are directly checking the value of `config.api_only`", I for one will be very sad.